### PR TITLE
[WIP]Added some scene tests

### PR DESCRIPTION
**Pull request by tarzzz**
Added some basic tests for scene. the tests for json generation and display(), are not included.

### CODE REVIEW

File: tests/test_visualization_frame_scene.py [hunk 1]
```diff
@@ -165,26 +166,55 @@ def test_camera(self):
...
         #unnamed camera
-        camera1 = Camera(I,O)
+        camera1 = Camera(self.I, self.O)
         assert camera1.name == 'unnamed'
-        assert camera1.reference_frame == I
-        assert cameta1.origin == O
+        assert camera1.reference_frame == self.I
+        assert camera1.origin == self.O
+
+    def test_scene(self):
```
Comment (tarzzz) [inline tests/test_visualization_frame_scene.py:193]: @moorepants @fitze Please have a look at it and comment.

File: tests/test_visualization_frame_scene.py [hunk 2]
```diff
@@ -165,26 +166,55 @@ def test_camera(self):
...
+        
+        self.scene.remove_frame(self.frame1)
+        assert self.scene.frames[0] is self.frame2
+        assert self.scene.frames[1] is self.frame3
+        
+        self.scene.add_visualization_frame(self.frame1)
+        assert self.scene.frames[0] is self.frame1
+        assert self.scene.frames[1] is self.frame2
+        assert self.scene.frames[2] is self.frame3
+        #TODO check for multiple frame insertion
```
Comment (chrisdembia) [inline tests/test_visualization_frame_scene.py:219]: How will you do this?

**Comment by chrisdembia** [comment]
Cool. What about JSON-related methods? Will those be elsewhere, or here?

File: tests/test_visualization_frame_scene.py (hunk 2 above)
Comment (tarzzz) [inline tests/test_visualization_frame_scene.py:219]: well its more like ..
self.scene.add_visualization_frames(fram1,frame2,.. etc.)
nothing much

**Comment by tarzzz** [comment]
Regarding JSON like methods, Jason and I have decided to put up an extended and detailed example code. most probably the three link conical pendulum, or something whose results we are aware of.
Then we will use that example to check JSON generation and numerical integration, Basically stuff which return bulk values to handle. 
The tests written here, are kept simple, clean and basic only

**Comment by tarzzz** [comment]
I am merging this pull request, to change the directory structure, and then we can continue the discussion on another thread, or emails maybe.

**Pull request merged**